### PR TITLE
Identify XOR Encoded PE files

### LINKED ISSUE #757: identify embedded XOR encoded PE files and render those
extension of #756

### PR DESCRIPTION
**Pull request by ooprathamm**
Closes #757 

Taking reference from how capa identifies xor encodeded [[features/extractors/helper.py](https://github.com/mandiant/capa/blob/master/capa/features/extractors/helpers.py)]

image

image

### CODE REVIEW

**Review by williballenthin — COMMENTED** [review]
really neat! can we somehow name the region to indicate the data was decoded and the key used?

**Comment by ooprathamm** [comment]
* Rendering PE is XOR decoded and key in the header
image

**Comment by williballenthin** [comment]
Implementation looks great. 

Question for the room ( @mr-tz ): this mode now prints strings that aren't visible verbatim from strings.exe. Is that ok?

**Review by williballenthin — APPROVED** [review]

**Comment by mr-tz** [comment]
Cool! Should we add a new tag like `#floss` or `#decoded`?

**Comment by mr-tz** [comment]
How worried are we about performance here? Can you provide some stats about how long this runs for a small and a huge binary?

**Comment by williballenthin** [comment]
> #decoded

This sounds good to me!

**Comment by ooprathamm** [comment]
@mr-tz I did some pretty basic profiling on the compute_layout() 
* Metric: Execution time
* Each call was performed 15 times taking avg value for calculation of total_time of compute_layout for each file
image

* The results were pretty inconsistent across multiple runs, showing only lowermost and uppermost values

**Comment by ooprathamm** [comment]
* #decoded tag when strings are xor_decoded

image

**Comment by mr-tz** [comment]
Nice, that looks good to me. We could discuss `#decoded` vs. a better term further, but this works for now.

Weird that the install fails, can we revert to lancelot 0.8.7 to see CI succeed?

**Comment by ooprathamm** [comment]
@williballenthin can I make a commit downgrading lancelot ?

**Comment by williballenthin** [comment]
sounds good to me. i won't be able to fix lancelot for maybe a week or so.

**Review by mr-tz — APPROVED** [review]
lgtm, thanks!

File: floss/qs/main.py [hunk 1]
```diff
@@ -1051,6 +1089,8 @@ def render_strings(
     BORDER_STYLE = MUTED_STYLE
 
     name = layout.name
+    if isinstance(layout, PELayout) and layout.xor_key:  # Check if the layout is PELayout and is xored
+        name += f" (XOR decoded with key: {layout.xor_key})"
```
Comment (mr-tz) [inline floss/qs/main.py:1093]:
```suggestion
        name += f" (XOR decoded with key: 0x{layout.xor_key:x})"
```

Comment (mr-tz) [reply floss/qs/main.py:1093]: last little tweak

**Pull request merged**